metrics/open-tracing-backend/src/main/scala/sttp/client/opentracing/OpenTracingBackend.scala

```diff
@@ -19,14 +20,22 @@ class OpenTracingBackend[F[_], S] private (delegate: SttpBackend[F, S, NothingT]
   override def send[T](request: Request[T, S]): F[Response[T]] =
     responseMonad
       .eval {
-        val span = tracer
-          .buildSpan(
-            request
-              .tag(OpenTracingBackend.OperationIdRequestTag)
-              .getOrElse("default-operation-id")
-              .toString
-          )
-          .withTag(Tags.SPAN_KIND, Tags.SPAN_KIND_CLIENT)
+        val spanBuilderTransformer: SpanBuilderTransformer =
+          request
+            .tag(OpenTracingBackend.SpanBuilderTransformerRequestTag)
+            .collectFirst {
+              case f: SpanBuilderTransformer => f
```

Comment:
This is suboptimal, because any `Function1[_, _]` will match here, but we do same for `SpanTransformer` itself.